src/site/content/en/angular/precaching-with-the-angular-service-worker/index.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: 'Precaching with the Angular service worker'
+subhead: |
+  Use the Angular service worker to make your app faster and more reliable on networks with poor connectivity!
+hero: performance-budgets-with-the-angular-cli.jpg
+date: 2019-07-02
+description: |
+  Learn how to use the Angular service worker to precache the static assets in your app.
+authors:
+  - mgechev
+---
+
+## Dealing with limited connectivity
+
+When users have limited network access—or none at all—web app functionality can significantly degrade and often fails. Using a [service worker](https://developers.google.com/web/fundamentals/primers/service-workers/) to provide precaching lets you intercept network requests and deliver responses directly from a local cache instead of retrieving them from the network. Once your app’s assets have been cached, this approach can really speed up an app and make it work when the user is offline.
```

Comment:
```suggestion
When users have limited network access—or none at all—web app functionality can significantly degrade and often fails. Using a [service worker](https://developers.google.com/web/fundamentals/primers/service-workers/) to provide precaching lets you intercept network requests and deliver responses directly from a local cache instead of retrieving them from the network. Once your app's assets have been cached, this approach can really speed up an app and make it work when the user is offline.
```